invertedai_cpp/invertedai/blame_request.cc

```diff
@@ -0,0 +1,169 @@
+        TrafficLightState traffic_light_state = {element[0], element[1]};
+        traffic_light_states.push_back(traffic_light_state);
+      }
+      traffic_light_state_history.push_back(traffic_light_states);
+    }
+    this->traffic_light_state_history_ = traffic_light_state_history;
+  }
+  this->get_birdviews_ = this->body_json_["get_birdviews"].is_boolean()
+                             ? this->body_json_["get_birdviews"].get<bool>()
+                             : false;
+  this->get_reasons_ = this->body_json_["get_reasons"].is_boolean()
+                           ? this->body_json_["get_reasons"].get<bool>()
+                           : false;
+  this->get_confidence_score_ =
+      this->body_json_["get_confidence_score"].is_boolean()
+          ? this->body_json_["get_confidence_score"].get<bool>()
+          : false;
+}
+
+void BlameRequest::refresh_body_json_() {
+  this->body_json_["location"] = this->location_;
+  this->body_json_["colliding_agents"] = this->colliding_agents_;
+  this->body_json_["agent_state_history"].clear();
+  for (const std::vector<AgentState> &agent_states :
+       this->agent_state_history_) {
+    json elements;
+    elements.clear();
+    for (const AgentState &agent_state : agent_states) {
+      json element = {agent_state.x, agent_state.y, agent_state.orientation,
+                      agent_state.speed};
+      elements.push_back(element);
+    }
+    this->body_json_["agent_state_history"].push_back(elements);
+  }
+  this->body_json_["agent_attributes"].clear();
+  for (const AgentAttributes &agent_attribute : this->agent_attributes_) {
+    json element = {agent_attribute.length, agent_attribute.width,
+                    agent_attribute.rear_axis_offset};
+    this->body_json_["agent_attributes"].push_back(element);
+  }
+  if (this->traffic_light_state_history_.has_value()) {
+    this->body_json_["traffic_light_state_history"].clear();
+    for (const std::vector<TrafficLightState> &traffic_light_states :
+         this->traffic_light_state_history_.value()) {
+      json elements;
+      elements.clear();
+      for (const TrafficLightState &traffic_light_state :
+           traffic_light_states) {
+        json element = {traffic_light_state.id, traffic_light_state.value};
+        elements.push_back(element);
+      }
+      this->body_json_["traffic_light_state_history"].push_back(elements);
+    }
+  } else {
+    this->body_json_["traffic_light_state_history"] = nullptr;
+  }
+  this->body_json_["get_birdviews"] = this->get_birdviews_;
+  this->body_json_["get_reasons"] = this->get_reasons_;
+  this->body_json_["get_confidence_score"] = this->get_confidence_score_;
+}
+
+std::string BlameRequest::body_str() {
+  this->refresh_body_json_();
+  return this->body_json_.dump();
+}
+
+std::string BlameRequest::location() const { return this->location_; }
+
+std::pair<int, int> BlameRequest::colliding_agents() const {
+  return this->colliding_agents_;
+}
```

Comment:
Would it possibly be wise to make this a "std::vector" for future proofing reasons in case we want to be more flexible with the number of agents involved in a collision or do you think this is unnecessary?

Reply:
I want to keep consistent with the Python API. We can change it as needed in the future.
![image](https://github.com/inverted-ai/invertedai/assets/26733898/0362a424-52f3-41de-834f-cd583976fba7)